DirectEmployers/MyJobs
# PD-3451 Fix discrepancies in timeline

## What's this PR do?
Fixing glitches mentioned in PD-3451

## Screenshots (if appropriate)
**Before**: data were aggregated based on UTC, so a _day_ is UTC 00:00 - 23:59. That means for someone in New York, it's 19:00 to 18:59  (or 20:00 to 19:59 during daylight savings), that's why there is a "tail" at the end (extra 4-5 hours till Eastern midnight but next day for UTC)

![image](https://user-images.githubusercontent.com/27963853/27975154-0cdd5b08-632f-11e7-9e3b-f55268719208.png)

**After**: each data point now represents total views of a day _in client timezone_, e.g. for someone in New York, it aggregates UTC 04:00am to 03:59am the next day.

![image](https://user-images.githubusercontent.com/27963853/27975170-21375676-632f-11e7-9bba-53647faa1fad.png)

## Where should the reviewer start?
`myreports/models.py`, line#456 , logic for timeline data formatting (previously `format_timeline_data`in analytics/api.py) has been moved here
`myreports/datasources` there is a small change in `job_views` and `analytics_event`, to shift UTC to default timezone before aggregating

## How should this be manually tested?
Visit any report page with timeline, select a custom date range, you should expect the same behavior in the screenshots (timeline ending at right date and all days without data represented with zeroes). Timeline graphs exported in Excel also reflect this change.

![image](https://user-images.githubusercontent.com/27963853/27975563-c483716a-6330-11e7-8342-bebd4174414f.png)

## Any background context you want to provide?
No front end changes required.
Right now, USE_TZ is enabled for our app, which means naïve time objects will default to time zone: `America/New_York`. 
We should probably start using offset-aware timestamps everywhere. Include client time zone information in requests, this will enable localizing time series data for users in different timezones. To do that, we need to make sure front end always passes client timezone, and we do not lose that information during date re-formatting (where time gets truncated): everything that gets passed into `DateRangeFilter` (myreports/datasources/util.py) should be timezone aware (right now it's not the case).
